### DISCUSSION TITLE
Stripe Apple Pay Button Not working

### DISCUSSION THREAD

**Discussion by Ahmad0Tanveer** [flutter-stripe/flutter_stripe#1571] [Q&A] [answered]
in android GOOGLe Pay button functions correctly also Stripe  work correctly. but in IOS stripe working perfectly
buttton Apple Play button Just crashing the close the Sheet and not show any error warning also
and in simulator its working perfeclty
```
 await Stripe.instance.initPaymentSheet(
          paymentSheetParameters: SetupPaymentSheetParameters(
              paymentIntentClientSecret: paymentIntentData!['client_secret'],
              billingDetails: billingDetails,
              applePay: 
                PaymentSheetApplePay(merchantCountryCode: 'USD', 
              ),
              googlePay: PaymentSheetGooglePay(
                  merchantCountryCode: 'USD', currencyCode: "USA"),
              style: ThemeMode.light,
              merchantDisplayName: 'Compnay Name'));
     
    } catch (e) {
      Get.snackbar("title", "${e}", duration: Duration(seconds: 60));
      print('exception:$e');
    }
```

    
 Event is **merchantIdentifier**  
 Certificate  and merchantId created and also uploaded to Stripe Dashboard but stil not working
in stipe Payment Faild message is 
"The customer has not entered their payment method"

**Comment by remonh87** [comment]
did you have an apple pay method configured on the phone and did it work on a simulator? Normally if you run it in xcode the Stripe sdk will provide more finer grained logging.

**Reply by Ahmad0Tanveer** [reply]
yes its work on simulators certficates and merhent id is added also

**Reply by remonh87** [reply]
I would recommend checking https://support.stripe.com/questions/enable-apple-pay-on-your-stripe-account if you configured it correctly. The fact that it works on simulators shows me that the flutter implementation is fine. Else I would recommend contacting stripe support

**Comment by developermvuk** [comment]
can someone help? Payment sheet showing when we press apple pay icon to pay says processing and closes the payment sheet. Device has working cards registered to apple pay but its not showing the cards on pop up and only closing the sheet?

**Reply by remonh87** [reply]
I would recommend checking the xcode logs. Also make sure that everything is configured properly. You need to make sure the certificate is uploaded to stripe: https://support.stripe.com/questions/enable-apple-pay-on-your-stripe-account

**Comment by mahimrocky** [comment, marked as answer]
Hi @Ahmad0Tanveer did you solve that issue?
